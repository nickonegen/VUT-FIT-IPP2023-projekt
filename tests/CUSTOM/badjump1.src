# Sample program with unreachable jumps
.IPPcode23

# Infinite loop
# DEFVAR GF@result # undefined variable, parser should still pass
LABEL loop
ADD GF@result int@1 int@1
JUMP loop # 1 backjump

# These jumps are not reachable
JUMP non_existant_label_1
JUMP non_existant_label_2