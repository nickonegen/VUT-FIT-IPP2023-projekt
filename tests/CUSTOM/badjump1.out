<?xml version="1.0" encoding="UTF-8"?>
<program language="IPPcode23"><instruction order="1" opcode="LABEL"><arg1 type="label">loop</arg1></instruction><instruction order="2" opcode="ADD"><arg1 type="var">GF@result</arg1><arg2 type="int">1</arg2><arg3 type="int">1</arg3></instruction><instruction order="3" opcode="JUMP"><arg1 type="label">loop</arg1></instruction><instruction order="4" opcode="JUMP"><arg1 type="label">non_existant_label_1</arg1></instruction><instruction order="5" opcode="JUMP"><arg1 type="label">non_existant_label_2</arg1></instruction></program>
